File: src/Moryx.AbstractionLayer.Products.Endpoints/Model/ProductModel.cs
// Copyright (c) 2023, Phoenix Contact GmbH & Co. KG
// Licensed under the Apache License, Version 2.0

using System.Runtime.Serialization;
using Moryx.Serialization;

namespace Moryx.AbstractionLayer.Products.Endpoints
{
    [DataContract]
    public class ProductModel
    {
        [DataMember]
        public long Id { get; set; }

        [DataMember]
        public string Name { get; set; }

        [DataMember]
        public ProductState State { get; set; }

        [DataMember]
        public string Identifier { get; set; }

        [DataMember]
        public short Revision { get; set; }

        [DataMember]
        public string Type { get; set; }

        [DataMember]
        public Entry Properties { get; set; }      

        [DataMember]
        public ProductFileModel[] Files { get; set; }

        [DataMember]
        public PartConnector[] Parts { get; set; }

        [DataMember]
        public RecipeModel[] Recipes { get; set; }
    }
}

File: src/Moryx.Workplans.Web/src/app/api/models/product-model.ts
/* tslint:disable */
/* eslint-disable */
import { Entry } from '@moryx/ngx-web-framework';
import { PartConnector } from './part-connector';
import { ProductFileModel } from './product-file-model';
import { ProductState } from './product-state';
import { RecipeModel } from './recipe-model';
export interface ProductModel {
  files?: null | Array<ProductFileModel>;
  id?: number;
  identifier?: null | string;
  name?: null | string;
  parts?: null | Array<PartConnector>;
  properties?: Entry;
  recipes?: null | Array<RecipeModel>;
  revision?: number;
  state?: ProductState;
  type?: null | string;
}
